src/main/java/goblintserver/GoblintServer.java

```diff
@@ -64,13 +65,23 @@ public boolean startGoblintServer() {
                 return false;
             }
 
-            while(!goblintSocket.exists()) {
-                Thread.sleep(1000); 
+            // wait until Goblint socket is created before continuing
+            WatchService watchService = FileSystems.getDefault().newWatchService();
+            Path path = Paths.get(System.getProperty("user.dir"));
+            path.register(watchService, StandardWatchEventKinds.ENTRY_CREATE );
+            WatchKey key;
+            while ((key = watchService.take()) != null) {
+                for (WatchEvent<?> event : key.pollEvents()) {
+                    if (event.context().toString().equals(goblintSocket.toString())) {
```

Comment:
For whatever reason the API is a bit weird, but I think it should be safe to cast to `Path` and do a non-string check:
```suggestion
                    if (((Path) event.context()).equals(goblintSocket)) {
```